frontend/apps/web/app/(mgmt)/[account]/jobs/[id]/destinations/components/DestinationConnectionCard.tsx

```diff
@@ -52,19 +68,62 @@ export default function DestinationConnectionCard({
   mutate,
   isDeleteDisabled,
   jobSourceId,
+  jobmappings,
 }: Props): ReactElement {
+  const { account } = useAccount();
   const { mutateAsync: setJobDestConnection } = useMutation(
     JobService.method.updateJobDestinationConnection
   );
   const { mutateAsync: removeJobDestConnection } = useMutation(
     JobService.method.deleteJobDestinationConnection
   );
 
+  const { mutateAsync: validateJobMappingsAsync } = useMutation(
+    JobService.method.validateJobMappings
+  );
+
+  const [validateMappingsResponse, setValidateMappingsResponse] = useState<
+    ValidateJobMappingsResponse | undefined
+  >();
+
   const form = useForm({
     resolver: yupResolver<NewDestinationFormValues>(NewDestinationFormValues),
     values: getDestinationFormValuesOrDefaultFromDestination(destination),
   });
 
+  const jobmappingsLength = jobmappings?.length || 0;
+
+  async function validateMappings(connectionId: string) {
+    if (!jobmappings || jobmappings.length == 0) {
+      return;
+    }
+    try {
+      const body = create(ValidateJobMappingsRequestSchema, {
+        accountId: account?.id,
+        mappings: jobmappings,
+        virtualForeignKeys: [],
+        connectionId: connectionId,
+      });
+      const res = await validateJobMappingsAsync(body);
+      setValidateMappingsResponse(res);
+    } catch (error) {
+      console.error('Failed to validate job mappings:', error);
+      toast.error('Unable to validate job mappings', {
+        description: getErrorMessage(error),
+      });
+    }
+  }
+
+  useEffect(() => {
+    if (!account?.id || !destination.connectionId || jobmappingsLength === 0) {
+      return;
+    }
+    const validateJobMappings = async () => {
+      await validateMappings(destination.connectionId);
+    };
+    validateJobMappings();
+  }, [account?.id, destination.connectionId, jobmappingsLength]);
```

Comment:
Hm, why isn't this just a `useQuery`? All queries can be conditionally enabled anyways, which is basically your if condition.

Reply:
Another thought is that we could also conditionally invoke only if the init table schema is not enabled? maybe that is unnecessary though.

Reply:
copy paste 😭